src/com/dojo/foundations/parking/vehicle/Vehicle.java

```diff
@@ -0,0 +1,19 @@
+package com.dojo.foundations.parking.vehicle;
+
+public abstract class Vehicle implements Billing {
```

Comment:
Excellent implementation, clean and clear.  The SOLID principles are being used wisely